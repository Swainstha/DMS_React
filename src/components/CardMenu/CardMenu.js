import React from 'react';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import IntlMessages from '../../util/IntlMessages';

export const options = [
  // <IntlMessages id="popup.updateData"/>,
  // <IntlMessages id="popup.detailedLog"/>,
  // <IntlMessages id="popup.statistics"/>,
  // <IntlMessages id="popup.clearData"/>
  "Update Data",
  "Detailed Log"
];
const CardMenu = (props) => {

  const {menuState, anchorEl, handleRequestClose} = props;
  return (
    <Menu id="long-menu" data-test="Menu-component"
          anchorEl={anchorEl}
          open={menuState}
          onClose={handleRequestClose}

          MenuListProps={{
            style: {
              width: 150,
              paddingTop: 0,
              paddingBottom: 0
            },
          }}>
      {options.map(option =>
        <MenuItem key={option} data-tests={option} onClick={() => handleRequestClose(option)} data-test="MenuItem-component">
          {option}
        </MenuItem>,
      )}
    </Menu>
  );
};

export default CardMenu;