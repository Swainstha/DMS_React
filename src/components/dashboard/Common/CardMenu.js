import React from 'react';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import IntlMessages from '../../../util/IntlMessages';

export const options = [
  <IntlMessages id="popup.updateData"/>,
  <IntlMessages id="popup.detailedLog"/>,
  <IntlMessages id="popup.statistics"/>,
  <IntlMessages id="popup.clearData"/>
];
const CardMenu = (props) => {


  const {menuState, anchorEl, handleRequestClose} = props;
  return (
    <Menu id="long-menu"
          anchorEl={anchorEl}
          open={menuState}
          onClose={handleRequestClose}

          MenuListProps={{
            style: {
              width: 150,
              paddingTop: 0,
              paddingBottom: 0
            },
          }} data-test="container-component">
      {options.map((option,index) =>
        <MenuItem key={option} onClick={handleRequestClose} data-test={`menu-component${index}`}>
          {option}
        </MenuItem>,
      )}
    </Menu>
  );
};

export default CardMenu;