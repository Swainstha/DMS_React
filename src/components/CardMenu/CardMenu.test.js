import React from 'react';
import CardMenu , {options} from './CardMenu';
import {
    configure,
    shallow,
} from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})
const function_click=jest.fn()

const setup = (props = {}) => {
    return shallow( < CardMenu {...props}  />
    )
}

const findByTestAttr=(wrapper, val)=>{
    return wrapper.find(`[data-test='${val}']`)
}

const findByTestKey=(wrapper, val)=>{
    return wrapper.find(`[data-tests='${val}']`)
}

it("CardMenu",()=>{
    const wrapper= setup({anchorEl:'', menuState:'', handleRequestClose:()=>{}});
    expect(wrapper).toMatchSnapshot();
})

it("CardMenu props",()=>{
    const wrapper= setup({anchorEl:'', menuState:'', handleRequestClose:function_click});

    let child0= findByTestAttr(wrapper,'Menu-component')
    let child1= findByTestAttr(wrapper,'MenuItem-component')
    expect(child0).toHaveLength(1)
    expect(child1).toHaveLength(options.length)
    let childOption1= findByTestKey(wrapper,options[0])
    childOption1.simulate('click');
    wrapper.update();
    expect(function_click).toHaveBeenCalledTimes(1)
    expect(function_click).toHaveBeenCalledWith(options[0])
    let childOption2= findByTestKey(wrapper,options[1])
    childOption2.simulate('click');
    wrapper.update();
    expect(function_click).toHaveBeenCalledTimes(2)
    expect(function_click).toHaveBeenCalledWith(options[1])


})
