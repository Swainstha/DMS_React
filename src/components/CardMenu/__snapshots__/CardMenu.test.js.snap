// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardMenu 1`] = `ShallowWrapper {}`;
